css-borders-4/Overview.bs

```diff
@@ -336,29 +336,171 @@ The 'border-radius' shorthand</h4>
+	horizontal radius and the values after the slash set the vertical radius.
+	If there is no slash, then the values set both radii equally.
+	The two values for the radii are given in the order
+	top-left, top-right for 'corner-top-shape',
+	top-right, bottom-right for 'corner-right-shape',
+	bottom-left, bottom-right for 'corner-bottom-shape',
+	top-left, bottom-left for 'corner-left-shape',
+	start-start, start-end for 'corner-block-start-shape',
+	end-start, end-end for 'corner-block-end-shape'
+	start-start, end-start for 'corner-inline-start-shape',
+	and start-end, end-end for 'corner-inline-end-shape'.
+	If the second value is omitted it is copied from the first.
+
+<h4 id=corner-shape-shorthand>
+Shaping all the corners at once</h4>
 
 	<pre class="propdef">
 		Name: corner-shape
-		Value: [ round | angle ]{1,4}
+		Value: <<corner-shape-value>>{1,2} / [ <<corner-shape-value>>{1,2} ]?
 		Initial: round
-		Applies to: all elements, except table element when 'border-collapse' is ''collapse''
+		Applies to: all elements where 'border-radius' can apply
 		Inherited: no
-		Animation type: discrete
+		Animation type: see individual properties
 	</pre>
 
-	By default, non-zero border-radii define
-	a quarter-ellipse that rounds the affected corners.
-	However in some cases, other corner shapes are desired.
-	The 'corner-shape' property specifies a reinterpretation of the radii
-	to define other corner shapes.
+	Applies the shape to all corners, following the same rules as ''border-radius''.
 
-	<dl dfn-type="value" dfn-for="corner-shape">
-		<dt><dfn>''round''</dfn>
-		<dd>Border radii define a convex elliptical curve at the corner.
-		<dt><dfn>''angle''</dfn>
-		<dd>Border radii define a diagonal slice at the corner.
-	</dl>
+<h4 id=corner-shape-interpolation>
+Interpolating corner shapes</h4>
+
+Since a <<corner-shape-value>> can always be expressed a ''superellipse()'' with an [=superellipse exponent=] variable, interpolating between two
+<<corner-shape-value>>s is done by interpolating the [=superellipse exponent=] itself.
+Since it's an exponent, interpolating it linearly would result in an effect where concave corners interpolate at a much higher velocity than convex corners.
+To balance that, the <dfn>superellipse interpolation</dfn> formula describes how a [=superellipse exponent=] is converted to a value between 0 and 1, and vice versa:
+
+<div algorithm="superellipse-exponent-to-interpolation-value">
+To interpolate a <<number [0,&infin;]>> |exponent| to an interpolation value between 0 and 1:
+	1. If |exponent| is 0, return 0.
+	1. If |exponent| is &infin;, return 1.
+	1. Return <code>1/(2^(1/|exponent|))</code>.
+
+To convert a <<number [0,1]>> |interpolationValue| back to a [=superellipse exponent=]:
+	1. If |interpolationValue| is 0, return 0.
+	1. If |interpolationValue| is 1, return &infin;.
+	1. Return <code>ln(0.5)/ln(|interpolationValue|)</code>.
+</div>
+
+Issue: resolve on this or another interpolation formula.
```

Comment:
I'm assuming we need to investigate if this looks good right?

Reply:
Yes, opened #11608. It would also be impacted by a change in what `k` means (#11609).